aw-datastore/src/datastore.rs

```diff
@@ -573,6 +629,26 @@ impl DatastoreInstance {
         bucket_id: &str,
         event: &Event,
     ) -> Result<(), DatastoreError> {
+
+        let app_exists = event.data.get("app");
```

Comment:
Copy-paste of the same code above. Better to put it in a generic function. Unittests also wouldn't hurt.

Reply:
Agreed, this should be made into a generic gunction  -- the second path was because in my original implementation I found there was the second path that also created data entries (heartbeat)...   So I duplicated the code to verify it would cover all the cases...  Revamping the code is fine and I will check into moving this up to a higher layer, I just want to make sure I don't miss any code paths that plugins or other entities can call that will generate this data....